### ISSUE TITLE
Flutter video player 0.10.9+2 cannot be initialized in real Android project

### ISSUE THREAD

**Issue by XinwenCheng** [flutter/flutter#56499]
> Sorry I didn't follow the recommended format, but I think I made this question clear.

Following the latest instruction of [video_player V0.10.9](https://pub.dev/packages/video_player), I got very strange situation, for the [same video url](https://captnotes.com/wp-content/uploads/2020/02/demo_video_02.mp4), it works on a brand new flutter project just for demo, but doesn't work in my real project, I mean I got initialization issue in my real **Android** project. And I've already double checked configurations of the both projects, all are the same.

**Both projects works on iPhone, and demo project works on Android phone, but real project doesn't work on the same Android phone.**

I put following reference to both pubspec.yaml file:
```
video_player: ">=0.10.9+2 <2.0.0"
```

In my demo project, it works correctly:
```dart
class _MyHomePageState extends State<MyHomePage> {
  final videoUrl =
      'https://captnotes.com/wp-content/uploads/2020/02/demo_video_02.mp4';

  VideoPlayerController videoPlayerController;

  @override
  void initState() {
    super.initState();

    print('^_^initState()');

    videoPlayerController = VideoPlayerController.network(videoUrl)
      ..initialize().then((_) {
        setState(() {
          videoPlayerController.play();
          print('^_^play()');
        });
      })
      ..addListener(() {
        print('Is playing: ${videoPlayerController.value.isPlaying}');
      });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: videoPlayerController.value.initialized
            ? AspectRatio(
                aspectRatio: videoPlayerController.value.aspectRatio,
                child: VideoPlayer(videoPlayerController))
            : Container(),
      ),
    );
  }
}
```

But in my real project, initialize() will always make **MissingPluginException** twice, one for method name ```init``` on channel ```flutter.io/videoPlayer``` and one for method name ```create``` on channel ```flutter.io/videoPlayer```. This exception is thrown at line 319 in file ```platform_channel.dart```.
```dart
void initVideoPlayer() {
  if (TextUtil.isEmpty(videoUrl)) {
    logger.debug('videoUrl is EMPTY.');
    return;
  }

  logger.debug('videoUrl: $videoUrl');

  videoPlayerController?.dispose();
  videoPlayerController = VideoPlayerController.network(videoUrl)
    ..initialize()
    ..addListener(() {
      // logger.debug(
      //     'isVideoPlaying: $isVideoPlaying, videoPlayerController.value.isPlaying: ${videoPlayerController.value.isPlaying}');

      if (isVideoPreparing && videoPlayerController.value.isPlaying) {
        isVideoPreparing = false;
        isVideoPlaying = true;
      } else if (isVideoPlaying && !videoPlayerController.value.isPlaying) {
        videoPlayerController.seekTo(Duration(milliseconds: 0));
        isVideoPlaying = false;
        stopVideo();
        setState(() {});
      }
    });
}
```
[![Screenshot][1]][1]

I double checked, configurations of both projects are all the same including iOS and Android settings, absolutely follow every steps in official instruction.

I even thought if multidex makes this difference, but the demo project still works correctly after I set it to support multidex.

Any help would be very appreciated!

BTW, my flutter environment is on stable channel and healthy.
```
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.14.6 18G103, locale en-CN)
    • Flutter version 1.12.13+hotfix.9 at /Volumes/Transcend/Development/flutter
    • Framework revision f139b11009 (5 weeks ago), 2020-03-30 13:57:30 -0700
    • Engine revision af51afceb8
    • Dart version 2.7.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Volumes/Transcend/Development/AndroidSDK
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.9.0

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (1 available)
    • SM G9500 • 988a1b474239545746 • android-arm64 • Android 9 (API 28)

• No issues found!
```
  [1]: https://i.stack.imgur.com/sJX6C.png

**Comment by TahaTesser** [comment]
Hi @XinwenCheng 
can you please provide your  `flutter run --verbose` from your project while trying initialize the video_player? 
Thank you

**Comment by XinwenCheng** [comment]
Hi @TahaTesser ,

Unfortunately, after I upgraded flutter stable to V1.17.0, I cannot run my real project on my Android phone any more due to connection error as following logged:
```
Launching lib/main.dart on SM G9500 in debug mode...
Note: /Volumes/Transcend/Development/flutter/.pub-cache/hosted/pub.flutter-io.cn/data_plugin-0.0.17/android/src/main
/java/cn/bmob/data_plugin/util/InstallationUtils.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.                    
Note: Some input files use or override a deprecated API.                
Note: Recompile with -Xlint:deprecation for details.                    
Note: /Volumes/Transcend/Development/flutter/.pub-cache/hosted/pub.flutter-io.cn/url_launcher-5.4.5/android/src/main
/java/io/flutter/plugins/urllauncher/WebViewActivity.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.                    
Note: Some input files use or override a deprecated API.                
Note: Recompile with -Xlint:deprecation for details.                    
Note: /Volumes/Transcend/Development/flutter/.pub-cache/hosted/pub.flutter-io.cn/webview_flutter-0.3.21/android/src/
main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.                      
Note: /Volumes/Transcend/Development/flutter/.pub-cache/hosted/pub.flutter-io.cn/video_player-0.10.9+2/android/src/m
ain/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.                    
Running Gradle task 'assembleDebug'...                                  
                                                   
Running Gradle task 'assembleDebug'... Done                       145.5s (!)
✓ Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...                         6.4s
Error connecting to the service protocol: failed to connect to http://127.0.0.1:52077/-Ikghz1FUAg=/
```

What I've done for trying to solve this issue is:
1, Restart my phone;
2, Make sure I didn't use any VPN on my computer and phone;
3, Run '''flutter clean'''.

It still doesn't work, so I cannot provide you log of '''flutter run --verbose''' yet until I solved this problem.

And my result of '''flutter doctor -v''' now is:
```
[✓] Flutter (Channel stable, v1.17.0, on Mac OS X 10.14.6 18G103, locale en-CN)
    • Flutter version 1.17.0 at /Volumes/Transcend/Development/flutter
    • Framework revision e6b34c2b5c (5 days ago), 2020-05-02 11:39:18 -0700
    • Engine revision 540786dd51
    • Dart version 2.8.1

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Volumes/Transcend/Development/AndroidSDK
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.9.0

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (1 available)
    • SM G9500 • 988a1b474239545746 • android-arm64 • Android 9 (API 28)

• No issues found!
```

**Comment by XinwenCheng** [comment]
Hi @TahaTesser ,

Log of ```flutter run --verbose``` is attached.
[flutter_run_verbose.log](https://github.com/flutter/flutter/files/4596462/flutter_run_verbose.log)

**Comment by TahaTesser** [comment]
Hi @XinwenCheng 
Your logs doesn't have any exception and please created the error in the logs, try to initialiaze `video_player` and Also, to better address the issue, would be helpful
if you could post a minimal code sample to reproduce the problem
Thank you

**Comment by XinwenCheng** [comment]
Sorry, @TahaTesser , both of your request I cannot give your response yet. As I mentioned, after I upgraded to flutter V1.17.0, this project cannot be ran any more, and I'm still on this issue, so I cannot give your detailed log yet. And this issue only reproduces in my real project, if I create a new flutter project with video player, it works, and definitely I cannot provide your my real project.

If there's any other information I can provide you, please let me know how to get it, thanks very much in advance!

**Comment by TahaTesser** [comment]
Hi @XinwenCheng 
Without additional information, we are unfortunately not sure how to resolve this issue. We are
therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the
bug if you have any more information for us; we will reopen it right away!
Thanks for your contribution.

**Comment by XinwenCheng** [comment]
Hi @TahaTesser , seems this issue was solved since I upgrade video player to V0.10.10. And I also removed ```flutter_facebook_login: ^3.0.0``` from _pubspec.yaml_, not sure if this matters.
